Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/MainActivity.java

```diff
@@ -4,149 +4,121 @@
+  public void onPostCreate(Bundle savedInstanceState) {
     super.onPostCreate(savedInstanceState);
-    // Sync the toggle state after onRestoreInstanceState has occurred.
-    drawerToggle.syncState();
+    actionBarDrawerToggle.syncState();
   }
 
   @Override
   public void onConfigurationChanged(Configuration newConfig) {
     super.onConfigurationChanged(newConfig);
-    drawerToggle.onConfigurationChanged(newConfig);
+    actionBarDrawerToggle.onConfigurationChanged(newConfig);
   }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-    // Pass the event to ActionBarDrawerToggle, if it returns
-    // true, then it has handled the app icon touch event
-        /*if (drawerToggle.onOptionsItemSelected(item)) {
-            return true;
-        }*/
-    // Handle your other action bar items...
-    switch (item.getItemId()) {
-      // Respond to the action bar's Up/Home button
-      case android.R.id.home:
-        if (fragmentManager.getBackStackEntryCount() > 0) {
-          fragmentManager.popBackStack();
-        } else if (fragmentManager.getBackStackEntryCount() == 1) {
-          // to avoid looping below on initScreen
-          super.onBackPressed();
-          finish();
-        } else {
-          if (mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
-            mDrawerLayout.closeDrawer(GravityCompat.START);
-          }
-        }
-        return true;
-    }
-    return false;
-  }
-
 
-  @Override
-  public void onBackPressed() {
-    if (mDrawerLayout.isDrawerOpen(GravityCompat.START)) {
-      mDrawerLayout.closeDrawer(GravityCompat.START);
-      return;
+    // This is required to make the drawer toggle work
+    if (actionBarDrawerToggle.onOptionsItemSelected(item)) {
+      return true;
     }
 
-    super.onBackPressed();
+    return super.onOptionsItemSelected(item);
   }
 
+  private void replaceFragment(Fragment fragment, String title) {
+    updateToolbarTitle(title);
+    fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+  }
 
-  class DrawerItemClickListener implements ListView.OnItemClickListener {
-
-    @Override
-    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-
-      if (position < 3) {
-        ((NavigationAdapter) parent.getAdapter()).setSelectedIndex(position);
-      } //others are contact links
-      Uri data = null;
-      switch (position) {
-        case 0: //agenda
-          fragmentManager.beginTransaction().replace(R.id.fragment_container, new AgendaFragment())
-              .commit();
-          break;
-        case 1: //tweet
-          fragmentManager.beginTransaction().replace(R.id.fragment_container, new TweetsFragment())
-              .commit();
-          break;
-        case 2: // faq
-          fragmentManager.beginTransaction().replace(R.id.fragment_container, new FAQFragment()).commit();
-          break;
-        case 3: //contact, us
```

Comment:
where are all these links going? Seems like the only items left in the navigation are agenda/faq. Adding sections would be useful for separating social/agenda/contact links.

Reply:
As mentioned earlier smaller PR is expected thus there is a new [new branch](https://github.com/Droidcon-Boston/conference-app-android/tree/nishant/setup_social_screen) which adds an implementation of social links in a new Social Screen.

Once this got merged I was going to rebase and push it as a new PR.

Reply:
ok probably missed this all good